src/Doctrine/EntityClassGenerator.php

```diff
@@ -40,15 +40,17 @@ public function __construct(
     ) {
     }
 
-    public function generateEntityClass(ClassNameDetails $entityClassDetails, bool $apiResource, bool $withPasswordUpgrade = false, bool $generateRepositoryClass = true, bool $broadcast = false, EntityIdTypeEnum $useUuidIdentifier = EntityIdTypeEnum::INT): string
+    public function generateEntityClass(ClassNameDetails $entityClassDetails, bool $apiResource, bool $withPasswordUpgrade = false, bool $generateRepositoryClass = true, bool $broadcast = false, EntityIdTypeEnum $useUuidIdentifier = EntityIdTypeEnum::INT, ?string $tableName): string
     {
         $repoClassDetails = $this->generator->createClassNameDetails(
             $entityClassDetails->getRelativeName(),
             'Repository\\',
             'Repository'
         );
 
-        $tableName = $this->doctrineHelper->getPotentialTableName($entityClassDetails->getFullName());
+        if (null === $tableName) {
```

Comment:
Wouldn't you consider handling the most common case of pluralizing the class name somehow? Maybe via another option or something else? You can use [this](https://symfony.com/doc/current/string.html#inflector) Symfony inflector capability.

Reply:
This is a good point. But I think it's outside the scope of this PR, my purpose is to be able to explicitly choose the table name, not to change the default behavior. Both can also coexist, I think.

Reply:
I keep the idea to open a new PR in the next few days, or maybe you want to do it